pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -985,18 +983,36 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, allowEmul
 		}
 	}
 
-	ifaceNamingScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
+	vmConfigurator := netsetup.NewVMNetworkConfigurator(vmi, cache.CacheCreator{})
+	indexedInterfaces := netvmispec.IndexInterfaceSpecByName(vmi.Spec.Domain.Devices.Interfaces)
 	for _, network := range netvmispec.NetworksToHotplugWhosePodIfacesAreReady(vmi) {
 		log.Log.Infof("will hot plug %s", network.Name)
-		podIfaceName, wasFound := ifaceNamingScheme[network.Name]
+
+		ifaceToHotplug, wasFound := indexedInterfaces[network.Name]
 		if !wasFound {
-			return nil, fmt.Errorf("could not find the pod interface name for network [%s]", network.Name)
+			log.Log.Errorf("could not find a matching interface for network: %s", network.Name)
+			return nil, err
 		}
 
-		ifaceXML, err := xml.Marshal(converter.DomainInterfaceSpec(c, network.Name, virtnetlink.GenerateTapDeviceName(podIfaceName)))
+		domainInterfaces, err := converter.CreateDomainInterfaces(vmi, domain, c, false, ifaceToHotplug)
```

Comment:
The `false` stands for `virtioNetProhibited`, right?
On startup the converter has [logic](https://github.com/kubevirt/kubevirt/blob/583ab2daeb923f880c316d3ce567b40c398ae32d/pkg/virt-launcher/virtwrap/converter/converter.go#L1228) to calculate it. Why here on hotplug we can just assume it is `false`?

Reply:
Exposed a function so I can share the code.

I have honestly no idea is any of this works for interfaces that are *not* virtio though.

Reply:
Done.